### PR TITLE
Parallel inter-canister calls in Rust

### PR DESCRIPTION
**Pull request by oggy-dfin**
This example demonstrates how to use parallel inter-canister calls in Rust, and highlights the differences between parallel and sequential calls. A Motoko example will follow later.

### CODE REVIEW

File: rust/parallel_calls/README.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+[View this samples code on GitHub](https://github.com/dfinity/examples/tree/master/rust/parallel_calls).
+
+## Overview
+
+This example demonstrates how to implement inter-canister calls that run in parallel in Rust, and highlights some differences between parallel and sequential calls. Running independent calls in parallel can lower the latency, especially when messages are sent across subnets. For example, a canister that swaps two tokens might want to launch both token transfer operations in parallel.
+
+## Architecture
+
+The sample code revolves around two simple canisters, `caller` and `callee`. `Caller` has three endpoints:
+1. `setup_callee`, to set the ID of the callee canister
```
Comment (jessiemongeon1) [inline rust/parallel_calls/README.md:16]:
```suggestion
1. `setup_callee`, to set the ID of the callee canister.
```

**Review by jessiemongeon1 — APPROVED** [review]

File: rust/parallel_calls/src/callee/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[lib]
+crate-type = ["cdylib"]
+
+[dependencies]
+candid = "0.10"
+ic-cdk = "0.13"
+ic-cdk-timers = "0.7" # Feel free to remove this dependency if you don't need timers
```
Comment (dsarlis) [inline rust/parallel_calls/src/callee/Cargo.toml:14]: Do you need timers? I didn't see anything in the source files.

File: rust/parallel_calls/.gitignore [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# Various IDEs and Editors
```
Comment (dsarlis) [inline rust/parallel_calls/.gitignore:1]: Do you need a specific `.gitignore` for this example? We have one at the top level that should cover things I believe and no other examples have a specific `.gitignore`.

File: rust/parallel_calls/README.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
+---
+keywords: [advanced, rust, defi]
+---
+
+# Parallel inter-canister calls
+
+[View this samples code on GitHub](https://github.com/dfinity/examples/tree/master/rust/parallel_calls).
```
Comment (dsarlis) [inline rust/parallel_calls/README.md:7]:
```suggestion
[View this sample code on GitHub](https://github.com/dfinity/examples/tree/master/rust/parallel_calls).
```

File: rust/parallel_calls/src/multi_subnet/src/main.rs [hunk 5]
```diff
@@ -0,0 +1,78 @@
+use candid::{decode_one, encode_one, Principal};
+use pocket_ic::{PocketIcBuilder, WasmResult};
+use std::time::Instant;
+
+const INIT_CYCLES: u128 = 2_000_000_000_000;
```
Comment (dsarlis) [inline rust/parallel_calls/src/multi_subnet/src/main.rs:5]: nit: Leave an empty line below.

**Review by dsarlis — APPROVED** [review]

File: rust/parallel_calls/src/callee/Cargo.toml (hunk 2 above)
Comment (oggy-dfin) [reply rust/parallel_calls/src/callee/Cargo.toml:14]: Nope, thanks!

File: rust/parallel_calls/.gitignore (hunk 3 above)
Comment (oggy-dfin) [reply rust/parallel_calls/.gitignore:1]: Nice catch, thanks!

File: rust/parallel_calls/src/multi_subnet/src/main.rs (hunk 5 above)
Comment (oggy-dfin) [reply rust/parallel_calls/src/multi_subnet/src/main.rs:5]: Done

**Pull request merged**